CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2467,7 +2470,7 @@ protected function completeTransaction($result, $contributionID) {
         civicrm_api3('contribution', 'completetransaction', array(
           'id' => $contributionID,
           'trxn_id' => CRM_Utils_Array::value('trxn_id', $result),
-          'payment_processor_id' => $this->_paymentProcessor['id'],
+          'payment_processor_id' => CRM_Utils_Array::value('payment_processor_id', $result, $this->_paymentProcessor['id']),
```

Comment:
this line is the change - result has a processor id of 0 at this point so pass it on